p2p/protocol/identify/id.go

```diff
@@ -187,28 +186,58 @@ func (ids *IDService) ObservedAddrsFor(local ma.Multiaddr) []ma.Multiaddr {
 	return ids.observedAddrs.AddrsFor(local)
 }
 
+// IdentifyConn synchronously triggers an identify request on the connection and
+// waits for it to complete. If the connection is being identified by another
+// caller, this call will wait. If the connection has already been identified,
+// it will return immediately.
 func (ids *IDService) IdentifyConn(c network.Conn) {
+	<-ids.IdentifyWait(c)
+}
+
+// IdentifyWait triggers an identify (if the connection has not already been
+// identified) and returns a channel that is closed when the identify protocol
+// completes.
+func (ids *IDService) IdentifyWait(c network.Conn) <-chan struct{} {
```

Comment:
The name of this function is now a bit funky but this is the least-breaking way I could do it. Before, `IdentifyWait` would never trigger an identify operation. Now it does to be consistent everywhere.